source/common/network/listen_socket_impl.h

```diff
@@ -33,8 +33,10 @@ class ListenSocketImpl : public ListenSocket {
  */
 class TcpListenSocket : public ListenSocketImpl {
 public:
+  TcpListenSocket(const std::string& address, bool bind_to_port);
   TcpListenSocket(uint32_t port, bool bind_to_port);
```

Comment:
Why do we need the 2nd and 3rd constructors anymore? If for tests, can we fix tests to use address? If because of AddressPtr, can we make constructor take AddressPtr instead of string?